torch_xla/experimental/pjrt.py

```diff
@@ -99,14 +99,14 @@ def xla_device(n: Optional[int] = None,
   Returns:
     A `torch.device` representing an XLA device.
   """
+  if n is None:
+    return torch.device(torch_xla._XLAC._xla_get_default_device())
+
   devices = xm.get_xla_supported_devices(devkind=devkind)
-  device_index = n or (local_ordinal(default=0) % addressable_device_count())
-  if device_index > len(devices):
-    raise IndexError('Device index {} out of range in {}'.format(
-        device_index, devices))
+  if n > len(devices):
+    raise IndexError('Device index {} out of range in {}'.format(n, devices))
 
-  torch_xla._XLAC._xla_set_default_device(devices[device_index])
```

Comment:
any reason we remove this set default device logic?

Reply:
I set the default device in `run_thread_per_device` already, so it's not necessary to do it again here. IMO it's not intuitive for `xla_device` to more than return the _n_th or default device and actually affect thread-local state.

Reply:
was this line introduced by the pjrt or it existed in the xrt world too?

Reply:
It exists in XRT too:

https://github.com/pytorch/xla/blob/1715ccd0c11553c0e7ab6550715ae83769165b55/torch_xla/core/xla_model.py#L252

Reply:
I would say leave it as it is otherwise we might need to deal with weird xrt error(unless it is something blocking pjrt). If we want to refactor this from xrt it should be in a separate pr with corresponding test.

Reply:
`defualt_device` is useful in a lot of c++ code as well, we don't want to ask user to manually set default device so adding it to `xla_device` call is a compromise.

Reply:
Done